loopy/codegen/instruction.py

```diff
@@ -29,24 +29,29 @@
 from loopy.codegen import Unvectorizable
 from loopy.codegen.result import CodeGenerationResult
 from pymbolic.mapper.stringifier import PREC_NONE
+from pytools import memoize_method
+
+
+@memoize_method
+def _get_new_implemented_domain(chk_domain, implemented_domain):
```

Comment:
How come this is memoized on the `chk_domain`? (AKA, is there a reason not to do so on the kernel?)